### ISSUE TITLE
Sparse Model Benchmark

### ISSUE THREAD

**Issue by WeiHao97** [google/XNNPACK#404]
Dear authors,
In the "Fast Sparse ConvNets" paper, it says: "Instead, we implement a dense convolutional kernel which takes as input the image in the standard HWC layout and outputs the CHW layout consumed by the sparse operations in the rest of the network.' However, its seems to me that the layout of the feature maps are not changed after the Conv2d layer when I was inspecting the pre-trained sparse model with Netron. Could you please explain to me about this?

Another issue is, I encontered some problams when I tried to run the benchmark using bazel build with tensorflow lite on armv7 with linux (which I have raised an issue in their repo). And I am trying to run the end-2-end benchmark in this repository. Do you have c++ implementation of the pre-trained sparse model (like the ones in the models folder) so that I can run directly with this repo? Since its hard to extract the parameters used in the sparse model (subsampling size, relu ,etc).

Thank you very much!

**Comment by ekelsen** [comment]
The layout of the saved models is the layout that was used during training.  We modify the layout when loading for inference.

There does exist C++ code that runs the sparse models that used to generate the times in the paper, we do not consider it suitable for release due to its current lack of generality.  We're working on a more general solution.  Please reach out to the authors via email if you have a specific research use case that is being blocked by this.

**Comment by WeiHao97** [comment]
Hi @ekelsen,
> The layout of the saved models is the layout that was used during training. We modify the layout when loading for inference.

As I started to read tensorflow lite's code, I noticed that convolution operations are replaced by gemm kernels (in convolution-nhwc.c) rather then spmm kernels (in convolution-nchw.c). Is it because that the spmm are not integrated into the tf lite yet?

Thank you!

**Comment by ekelsen** [comment]
It is not integrated into TF-lite.  We will release a delegate that won't change the TF-lite code, but will essentially make TF-lite call XNNPACK to run a sparse model.

**Comment by WeiHao97** [comment]
Hi @ekelsen,
Is there a roughly expected date for the release? 

Thank you!

**Comment by ekelsen** [comment]
We hope to release it by CVPR.
